stdlib/string.mli

```diff
@@ -13,233 +13,329 @@
+    @raise Invalid_argument if [n < 0] or [n > ]{!Sys.max_string_length}. *)
+
+val init : int -> (int -> char) -> string
+(** [init n f] is a string of length [n] with index
+    [i] holding the character [f i] (called in increasing index order).
+
+    @raise Invalid_argument if [n < 0] or [n > ]{!Sys.max_string_length}.
+    @since 4.02.0 *)
 
 external length : string -> int = "%string_length"
-(** Return the length (number of characters) of the given string. *)
+(** [length s] is the length (number of bytes/characters) of [s]. *)
 
 external get : string -> int -> char = "%string_safe_get"
-(** [String.get s n] returns the character at index [n] in string [s].
-   You can also write [s.[n]] instead of [String.get s n].
-   @raise Invalid_argument if [n] not a valid index in [s]. *)
+(** [get s i] is the character at index [i] in [s]. This is the same
+    as writing [s.[i]].
 
+    @raise Invalid_argument if [i] not an index of [s]. *)
 
-external set : bytes -> int -> char -> unit = "%string_safe_set"
-  [@@ocaml.deprecated "Use Bytes.set instead."]
-(** [String.set s n c] modifies byte sequence [s] in place,
-   replacing the byte at index [n] with [c].
-   You can also write [s.[n] <- c] instead of [String.set s n c].
-   @raise Invalid_argument if [n] is not a valid index in [s].
+(** {1:concat Concatenating}
 
-   @deprecated This is a deprecated alias of {!Bytes.set}.[ ] *)
+    {b Note.} The {!Stdlib.( ^ )} binary operator concatenates two
+    strings. *)
 
-external create : int -> bytes = "caml_create_string"
-  [@@ocaml.deprecated "Use Bytes.create instead."]
-(** [String.create n] returns a fresh byte sequence of length [n].
-   The sequence is uninitialized and contains arbitrary bytes.
-   @raise Invalid_argument if [n < 0] or [n > ]{!Sys.max_string_length}.
+val concat : string -> string list -> string
+(** [concat sep ss] concatenates the list of strings [ss], inserting
+    the separator string [sep] between each.
 
-   @deprecated This is a deprecated alias of {!Bytes.create}.[ ] *)
+    @raise Invalid_argument if the result is longer than
+    {!Sys.max_string_length} bytes. *)
 
-val make : int -> char -> string
-(** [String.make n c] returns a fresh string of length [n],
-   filled with the character [c].
-   @raise Invalid_argument if [n < 0] or [n > ]{!Sys.max_string_length}. *)
+(** {1:predicates Predicates and comparisons} *)
 
-val init : int -> (int -> char) -> string
-(** [String.init n f] returns a string of length [n], with character
-    [i] initialized to the result of [f i] (called in increasing
-    index order).
+val equal : t -> t -> bool
+(** [equal s0 s1] is [true] iff [s0] and [s1] are character-wise equal.
+    @since 4.03.0 *)
 
-    @raise Invalid_argument if [n < 0] or [n > ]{!Sys.max_string_length}.
-    @since 4.02.0
-*)
+val compare : t -> t -> int
+(** [compare s0 s1] sorts [s0] and [s1] in lexicographical order. [compare]
+    behaves like {!Stdlib.compare} on strings but is more efficient. *)
```

Comment:
Splitting hairs, but AFAIU better efficiency is only in the case when type-based optimizations don't trigger.
```suggestion
    behaves like {!Stdlib.compare} on strings but may be more efficient. *)
```

Reply:
Reminds me I need a haircut. Changed as suggested.